mods/page_save.js

```diff
@@ -154,7 +172,8 @@ PageSave.prototype.saveWikitext = function(restbase, req) {
         return restbase.post({
             uri: new URI([rp.domain, 'sys', 'action', 'edit']),
             headers: {
-                cookie: req.headers.cookie
+                cookie: req.headers.cookie,
+                'x-forwarded-for': req.headers['x-forwarded-for']
```

Comment:
My preference would really be to let the action API handle it, so that such handling is uniform across clients.